pkg/aggregator/mocksender/asserts.go

```diff
@@ -108,13 +108,17 @@ func MatchEventLike(expected metrics.Event, allowedDelta time.Duration) interfac
 }
 
 // Compare an Event on specifics values:
-// AggregationKey, Priority, SourceTypeName, EventType and tag list
+// AggregationKey, Priority, SourceTypeName, EventType, Host, and tag list
 func eventLike(expectedEvent, actualEvent metrics.Event) bool {
 	if assert.ObjectsAreEqualValues(expectedEvent.AggregationKey, actualEvent.AggregationKey) &&
 		assert.ObjectsAreEqualValues(expectedEvent.Priority, actualEvent.Priority) &&
 		assert.ObjectsAreEqualValues(expectedEvent.SourceTypeName, actualEvent.SourceTypeName) &&
 		assert.ObjectsAreEqualValues(expectedEvent.EventType, actualEvent.EventType) &&
 		expectedInActual(expectedEvent.Tags, actualEvent.Tags) {
+		if expectedEvent.Host != "" {
+			// FIXME We only check the host entry if it has been set
```

Comment:
empty string `""` is a valid hostname value for events sent from checks. The aggregator will automatically add the agent hostname later.
I think you can safely `assert.ObjectsAreEqualValues(expectedEvent.Host, actualEvent.Host)` without any conditionnal (the default value for strings in structs if not specified is a `""`). Or did you run into issues?

Reply:
Ran into issues :-). Just adding this was causing some errors in some integration tests, which I spent some time to look into without success.
Also, I fear as they break other checks where the Host entry would have been forgotten, as it wasn't tested for.

What I had in mind is that not testing when `Host` was "" was already an improvement, and causes less issues. Here we could:
- Leave it like this (the FIXME should explain the thing)
- Add back the `assert.ObjectsAreEqualValues(expectedEvent.Host, actualEvent.Host)` and spend some more time on the broken tests

Reply:
Actually, I was afraid for nothing: the change was quite small to fix the broken tests :-)
Should be good to go!